tests/integration/test_signal_handling.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import inspect
+import time
+
+import pytest
+
+from dispatcher.publish import task
+
+
+@task(queue="test_channel")
+def sigterm_interceptor_task():
+    """
+    This function is top-level in an importable module.
+    The dispatcher sees e.g. "tests.integration.test_signal_handling.sigterm_interceptor_task".
+    """
+    import signal
+    import sys
+
+    def on_sigterm(signum, frame):
+        sys.stdout.write("SIGTERM was intercepted!\n")
+        sys.stdout.flush()
+
+    signal.signal(signal.SIGTERM, on_sigterm)
+    while True:
+        time.sleep(0.1)
+
+
+@pytest.mark.asyncio
+async def test_sigusr1_cancel_avoids_sigterm(apg_dispatcher, pg_control, test_settings, caplog):
+    """
+    Ensures dispatcher uses SIGUSR1, not SIGTERM. This test demonstrates canceling a task via SIGUSR1 without triggering
+    a custom SIGTERM handler.
+    """
+
+    # Verify code implementation uses SIGUSR1
+    from dispatcher.service.pool import PoolWorker
+    from dispatcher.worker.task import WorkerSignalHandler
+    code_init = inspect.getsource(WorkerSignalHandler.__init__)
+    code_cancel = inspect.getsource(PoolWorker.cancel)
+    assert "SIGUSR1" in code_init
+    assert "SIGUSR1" in code_cancel
```

Comment:
I'm cool with this, but this block of code would best be broken into another test. It could live in the `tests/unit/` folder, because it has no dependency on the fixtures or other systems.